### ISSUE TITLE
Form status not correctly preserved

### ISSUE THREAD

**Issue by danny0838** [gildas-lormeau/SingleFile#1524]
**Describe the bug**
The status of some form elements are not correctly preserved in the captured page.

**To Reproduce**
1. Create the test page and visit it:
    ```html
    <!DOCTYPE html>
    <meta charset="UTF-8">

    <input type="radio" value="a" data-checked='true'>aaa
    <input type="radio" value="b">bbb
    <input type="radio" value="c" checked>ccc
    <input type="radio" value="d" checked data-checked='false'>ddd

    <input type="checkbox" value="A" data-checked='true'>AAA
    <input type="checkbox" value="B">BBB
    <input type="checkbox" value="C" checked>CCC
    <input type="checkbox" value="D" checked data-checked='false'>DDD
    <input type="checkbox" value="E" checked data-indeterminate='true'>EEE

    <select>
    <option value="1" data-selected='true'>value: 1</option>
    <option value="2" selected>value: 2</option>
    </select>

    <script>
    for (const elem of document.querySelectorAll('*')) {
      for (const key in elem.dataset) {
        const value = JSON.parse(elem.dataset[key]);
        elem[key] = value;
      }
    }
    </script>
    ```
2. Save the page with default options
3. Open the saved page

**Expected behavior**
1. The radio button `ddd` is checked but should be unchecked.
2. The checkbox `DDD` is checked but should be unchecked.
3. The checkbox `EEE` is checked but should be indeterminate.
4. The option is value2 but should be value1.

**Environment**
- OS: Win10
- Browser: Chrome 127
- Version: 1.22.54

**Comment by gildas-lormeau** [comment]
Thank you. I fixed all the issues except the "indeterminate" state of `EEE`. I'm a little hesitant about this case because it requires the injection of a script.

**Comment by danny0838** [comment]
I understand your consideration. I don't like scripts either, but I am not against to handle script-driven matters with scripts, which is somehow of more fidelity of the original page.

Even for the the checked/selected/value attributes, reducing IDL into static HTML attribute breaks form resetting, which might be something the user wants to keep, although the form in the saved page is mostly not expected to be submittable.

WebScrapBook has 7 options for form status:

![圖片](https://github.com/user-attachments/assets/72625076-f58f-4421-98fc-f40c854b4fd0)

but I choose so because it's more expected that the saved page be editable and re-savable in WSB. After all SF is not WSB and you can do whatever you want. (BTW forget my broken English, maybe it should better be "states" rather than "status" here? 😆)

**Comment by gildas-lormeau** [comment]
Indeed, form management in SF is much more limited than in WSB. It's also the kind of problem I need to take some time to think about. The most important thing is that it's registered somewhere, and for that I thank you very much for creating this issue. You've described the problems perfectly (my english is broken too 😋).
